public/app/features/alerting/unified/components/rules/central-state-history/EventDetails.tsx

```diff
@@ -52,36 +57,103 @@ export function EventDetails({ record, logRecords }: EventDetailsProps) {
     );
   }
 
-  const getTransitionsCountByRuleUID = (ruleUID: string) => {
-    return logRecords.filter((record) => record.line.ruleUID === ruleUID).length;
-  };
-
   return (
     <Stack direction="column" gap={0.5}>
       <Stack direction={'row'} gap={6}>
-        <StateTransition record={record} />
+        <StateTransition record={record} addFilter={addFilter} />
         <ValueInTransition record={record} />
-        <NumberTransitions transitions={ruleUID ? getTransitionsCountByRuleUID(ruleUID) : 0} />
```

Comment:
we move this count logic to the `StateVisualization` component as it's where we fetch again by ruleUID all the log records for this alert rule to show a more accurate result.